geom/type_line_string.go

```diff
@@ -422,3 +423,19 @@ func (s LineString) Summary() string {
 func (s LineString) String() string {
 	return s.Summary()
 }
+
+// Simplify returns a simplified version of the LineString using the
+// Ramer-Douglas-Peucker algorithm. If the Ramer-Douglas-Peucker were to create
+// an invalid LineString (i.e. one having only a single distinct point), then
+// the empty LineString is returned.
+func (s LineString) Simplify(threshold float64) LineString {
+	seq := s.Coordinates()
+	floats := ramerDouglasPeucker(nil, seq, threshold)
+	seq = NewSequence(floats, seq.CoordinatesType())
+
+	simp, err := NewLineString(seq, OmitInvalid)
+	if err != nil {
+		panic("OmitInvalid used, so cannot panic: " + err.Error())
```

Comment:
Agreed, `"OmitInvalid used, unexpected error: "` is better here.

Reply:
> Is it because it's unexpected and should never happen?

Yes, that's right. Since it should never happen, it seems it seems unkind to users to require them to check an error. Perhaps there is a way to refactor things such that we don't have to panic on an error that should never happen, we might be able to add some internal LineString constructors that don't give an error. Leave it with me, I'll have a think of some better ways to do this.

Reply:
I've made a change to improve this in bc576058cf530e201d7e6dc507c5bfae423df2d2

It basically creates a new unexported `LineString` constructor that implicitly uses the `OmitEmpty` behaviour, and therefore _doesn't_ need to return an error. We can then call that directly from the `Simplify` method.